src/two_player_collision_example.cpp

```diff
@@ -211,73 +199,73 @@ TwoPlayerCollisionExample::TwoPlayerCollisionExample(
-      new QuadraticPolyline2Cost(kLaneCostWeight*0.1, lane3, {kP2XIdx, kP2YIdx},
-                                 "LaneCenter"));
-  const std::shared_ptr<SemiquadraticPolyline2Cost> p2_lane_r_cost(
-      new SemiquadraticPolyline2Cost(kLaneBoundaryCostWeight*0.1, lane3,
-                                     {kP2XIdx, kP2YIdx}, kLaneHalfWidth,
-                                     kOrientedRight, "LaneRightBoundary"));
-  const std::shared_ptr<SemiquadraticPolyline2Cost> p2_lane_l_cost(
-      new SemiquadraticPolyline2Cost(kLaneBoundaryCostWeight*0.1, lane3,
+  const std::shared_ptr<SemiquadraticPolyline2Cost> p2_lane1_l_cost(
+      new SemiquadraticPolyline2Cost(kLaneBoundaryCostWeight*10, lane1_p1p2,
                                      {kP2XIdx, kP2YIdx}, -kLaneHalfWidth,
                                      !kOrientedRight, "LaneLeftBoundary"));
-  p2_cost.AddStateCost(p2_lane_cost);
-  p2_cost.AddStateCost(p2_lane_r_cost);
-  p2_cost.AddStateCost(p2_lane_l_cost);
+  p1_cost.AddStateCost(p1_lane1_l_cost);
+  p2_cost.AddStateCost(p2_lane1_l_cost);
 
-  const std::shared_ptr<QuadraticPolyline2Cost> p2_lane_cost2(
-      new QuadraticPolyline2Cost(kLaneCostWeight*10, lane2, {kP2XIdx, kP2YIdx},
-                                 "LaneCenter"));
-  const std::shared_ptr<SemiquadraticPolyline2Cost> p2_lane_r_cost2(
-      new SemiquadraticPolyline2Cost(kLaneBoundaryCostWeight*10, lane2,
+  // p2 cost of leaving right lane boundary
+  const std::shared_ptr<SemiquadraticPolyline2Cost> p2_lane1_r_cost(
+      new SemiquadraticPolyline2Cost(kLaneBoundaryCostWeight, lane1_p1p2,
                                      {kP2XIdx, kP2YIdx}, kLaneHalfWidth,
                                      kOrientedRight, "LaneRightBoundary"));
-  const std::shared_ptr<SemiquadraticPolyline2Cost> p2_lane_l_cost2(
-      new SemiquadraticPolyline2Cost(kLaneBoundaryCostWeight*10, lane2,
-                                     {kP2XIdx, kP2YIdx}, -kLaneHalfWidth,
-                                     !kOrientedRight, "LaneLeftBoundary"));
-  p2_cost.AddStateCost(p2_lane_cost2);
-  p2_cost.AddStateCost(p2_lane_r_cost2);
-  p2_cost.AddStateCost(p2_lane_l_cost2);
+  p2_cost.AddStateCost(p2_lane1_r_cost);
+
+  // p1 right lane boundary cost
+  const Polyline2 lane1_p1(
+      {Point2(2.5 + kLaneHalfWidth, -50.0), Point2(2.5 + kLaneHalfWidth, -5.0)});
+  const Polyline2 lane2_p1(
+      {Point2(2.5 + kLaneHalfWidth, 5.0), Point2(2.5 + kLaneHalfWidth, 50.0)});
+  const Polyline2 lane3_p1(
+      {Point2(10.0, -5.0), Point2(10.0, 5.0)});
+  const Polyline2 lane4_p1(
+      {Point2(2.5 + kLaneHalfWidth, 5.0), Point2(25.0, 5.0)});
+  const Polyline2 lane5_p1(
+      {Point2(2.5 + kLaneHalfWidth, -5.0), Point2(25, -5.0)});
+  const std::shared_ptr<SemiquadraticPolyline2Cost> p1_lane2_r_cost(
+      new SemiquadraticPolyline2Cost(kLaneBoundaryCostWeight, lane1_p1,
+                                     {kP1XIdx, kP1YIdx}, 0,
```

Comment:
all 0's should be 0.0

Reply:
also you're reusing the "LaneRightBoundary" name. Give each cost for this player a unique name.

Reply:
done.